pkg/volume/glusterfs/glusterfs.go

```diff
@@ -439,9 +449,18 @@ func (d *glusterfsVolumeDeleter) Delete() error {
 		return err
 	}
 	d.provisioningConfig = *cfg
-
 	glog.V(4).Infof("glusterfs: deleting volume %q with configuration %+v", volumeId, d.provisioningConfig)
-
+	deleteGid := d.spec.Annotations["pv.beta.kubernetes.io/gid"]
+	scNameInSpec := d.spec.Annotations["volume.beta.kubernetes.io/storage-class"]
+	if deleteGid64, err := strconv.ParseUint(deleteGid, 10, 32); err == nil {
+		deleteGid32 = uint32(deleteGid64)
+	}
+	releaseBlock := Block{Start: deleteGid32, End: deleteGid32}
+	gidAllocator := d.plugin.gidallocatormap[scNameInSpec]
+	if err := gidAllocator.Release(releaseBlock); err != nil {
+		glog.Errorf("glusterfs:  error when releasing gid in storageclass: %s" , scNameInSpec)
+	}
```

Comment:
Suggestion: when the last GID from `gidAllocator` is released, i.e. there is no PV with given StorageClass, you should also remove whole `gidAllocator` from `d.plugin.gidallocatormap` so we don't end up with thousands of empty allocators.